proxy/http/HttpSM.cc

```diff
@@ -4874,6 +4874,8 @@ HttpSM::get_outbound_sni() const
     int len;
     char const *ptr = t_state.hdr_info.server_request.host_get(&len);
     zret.assign(ptr, len);
+  } else if (ua_txn && !strcmp(policy, "server_name"_tv)) {
+    zret.assign(ua_txn->get_netvc()->get_server_name(), ts::TextView::npos);
```

Comment:
Hmmm, are we actually strcmp()'ing these policy strings (from the yaml) on every request??

Reply:
If the policy is not empty, unfortunately, yes. I agree with we should avoid these `strcmp` with fixed strings except `@...` format.

https://docs.trafficserver.apache.org/en/latest/admin-guide/files/records.config.en.html?highlight=proxy.config.ssl.client.sni_policy#proxy-config-ssl-client-sni-policy